query_test.go

```diff
@@ -406,6 +406,15 @@ func TestQueryEffectiveBufferSize(t *testing.T) {
 	_, err = query.SetCoordinates(coordsWrite)
 	assert.Nil(t, err)
 
+	// Check the buffer sizes
+	offsetSize, dataSize, err := query.BufferSizeVar("a1")
+	assert.Nil(t, err)
+	assert.Equal(t, len(a1OffWrite) * 8, int(offsetSize))
```

Comment:
You can use `EqualValues` to test ignoring the datatypes, just an fyi.

Reply:
In this case I didn't know what the right way to do `sizeof` in go was -- the buffer size is returned in bytes.

Reply:
Actually I guess the right thing to do is return num elements instead of bytes with these getters. Will update the PR.

Reply:
You can replace 8 with `unsafe.Sizeof(a1OffWrite[0])`. Since a `slice` in go is a wrapper structure (like vector), you can't find the size directly, so you still need to multiple by the length.